### ISSUE TITLE
Refactor ProductImageMap and add the option to fetch products by remoteProductId  

### ISSUE THREAD

**Issue by anitaa1990** [woocommerce/woocommerce-android#3053]
While reviewing this PR, I noticed that we are using `ProductImageMap` to fetch the product details using the `remoteProductId`. This logic is called from adapter classes, which means it is called for each item in the adapter list.

It would be great if we could perhaps:
- Add logic to fetch the product details for a list of productIds, instead of fetching the one product at a time.
- Add this logic to be called from the `ViewModel`, rather than initiating this from the adapter class.

**Comment by dangermattic** [comment]
🚫 Please add a type label (e.g. **type: enhancement**) and a feature label (e.g. **feature: stats**) to this issue.

**Comment by JorgeMucientes** [comment]
I don't think this applies anymore. In any case this refactor proposal has been stale for more than 5 years. I don't expect we'll ever prioritise this.
